Edits for Jamstack 2021 Chapter (#2634)
HTTPArchive/almanac.httparchive.org: src/content/en/2021/jamstack.md

  sql_file="adoption_by_region.sql"
) }}

The number of sites in Europe for desktop is 23k versus mobile 26k, which is 1.1% of all web sites in that region. In the Americas, there are 26k desktop sites and 24k mobiles sites (1.2% of sites). Asia has almost the same numbers with 21k desktop and 22k mobile as the leading region with greater Jamstack adoption at 1.45%. Oceania and Africa have way lower overall numbers, but they have way greater Jamstack adoption. Oceania 2.19% and Africa 2%. Overall site adoption is at 1.1%.

> {# TODO - do you need to list the numbers for each, or should percentages be enough? or numbers for just the first as a baseline #}

Reviewer: We generally try to avoid absolute numbers and prefer percentages. The make no sense without context of how many they are out of.

However sometimes they can be useful to set context - particularly for small numbers (e.g. "it's only 0.1% but in the context of the web that's still a massive number of 600,000 sites"). But that should be the exception and not over used.

  height=794
) }}

> It's no surprise that first in the list is Northern America and USA location and driver of technologies. Most companies who invented SSGs are in the USA. But surprisingly, Western Europe is in second place and has 10k SSG mobile sites versus 17k for Northern America, but almost the same adoption level at 1.1%.
> 
> {# TODO - first sentence confuses me #}

Reviewer: ```suggestion
The list is ordered by the total number of SSG sites, but shows those as a percentage of all sites in that region. It's no surprise that top of the list is Northern America as most companies who invented SSGs are in the USA. However, as a percentage of all sites they are a a lower regions with only 1.1% of sites having adopted Jamstack. But surprisingly, Western Europe is in second place and has a similar low percentage adoption compared to some of the sub regions further down the list.
```

The tail also shows great results. Subregions with lower number of sites in general adopt technology even broader, like 4.8% of Micronesia sites.

### Summary of adoption by regions

SSG technology is adopted not only by places where it's invented and popularized, but also all around the world. Moreover, in places where the next generation of the web will be built, it grows even faster, mostly 2 times and sometimes 4 times, compared to others.

> {# TODO - does this need a source? ^ #}

Reviewer: I think it is sourced in the chart above. But not loving the absolute statement that this is where the next generation is to be built.

To be honest I'd drop this "Summary of adoption by regions" section. Don't think it adds much.

Reviewer: Dropped it in latest edit.

Jamstack CDN services are not just for network delivery. They provide a lot of functionality to allow developers to easy deploy and manage Jamstack sites. For example, Netlify provide easy to use functionality to deploy sites in scope of their service so developers can just update the code and the continuous deployment process is managed for them. <a hreflang="en" href="https://bejamas.io/compare/netlify-vs-vercel/">Jamstack CDNs provide many other features</a> such as _serverless functions_, _A/B testing_ etc.
  

> On the other hand, Cloudflare, Akamai, AWS are not only used for pure serving as CDN either, but can also provide protection service, DNS balancing and more. However, since we can't detect how exactly Cloudflare, Akamai, and AWS are used, results could be false positives if we look at them as Jamstack enablers, since the "Jamstack" part could be handled on origin servers and so not actually on these services.
> 
> {# TODO - can we divide that last sentence into 2 so it's not a run on? #}

Reviewer: ```suggestion
On the other hand, Cloudflare, Akamai, AWS are not only used purely for content deliver either, but can also provide protection service, DNS balancing and more. However, since we can't detect how exactly Cloudflare, Akamai, and AWS are used, results could be false positives if we look at them as Jamstack enablers. The "Jamstack" part could be handled on origin servers and so not actually on these services.
```

  sql_file="core_web_vitals_distribution.sql"
) }}

{# TODO - this figure description should be described more #}

> Next.js, is the most popular, mostly served by Cloudflare, Vercel, and AWS. Most of Gatsby's sites use Netlify, AWS, and Cloudflare.
> Nuxt.js sites preferred to be served by Cloudflare, AWS, and Netlify. Hugo mostly uses Netlify, and Jekyll as no surprise mostly on GitHub.
> 
> {# TODO - I did not understand the last sentence, is the comma used correctly? #}

Reviewer: ```suggestion
Next.js, is the most popular, mostly served by Cloudflare, Vercel, and AWS. Most of Gatsby's sites use Netlify, AWS, and Cloudflare. Nuxt.js sites preferred to be served by Cloudflare, AWS, and Netlify. Hugo mostly uses Netlify, and it's no surprised that Jekyll is used mostly on GitHub.
```

Why's it no surprise btw? Is that the only one really supported by GitHub? Did GitHub invent Jekyll? Or some other reason?

  sql_file="median_lighthouse_score.sql"
) }}

{# TODO - this figure description should be described more #}

> Median performance score for all SSGs across mobile varies. The top 3 SSGs with data can't even pass 40 score. Since they are pretty much popular in top ranking sites, and distributed all around the world, as we saw on data before, we can assumed that they are used across many different devices and networks. We can expect more out-of-the-box improvements like <a hreflang="en" href="https://nextjs.org/docs/basic-features/image-optimization">Next.js image component</a> to help performance.
> 
> {# TODO - there needs to be some grammar fix in the paragraph above but I didn't understand it #}

Reviewer: ```suggestion
The median performance score for all SSGs across mobile varies. The top 3 SSGs with by popularity can't even surpass a score of 40. Since they are used in top ranking sites and since users a likely distributed all around the world, we can assumed that they are used across many different devices and networks. We can expect more out-of-the-box improvements like <a hreflang="en" href="https://nextjs.org/docs/basic-features/image-optimization">Next.js image component</a> to help performance.
```

  sql_file="resource_weights.sql"
) }}

{# TODO - this figure description should be described more #}

On the other hand, CSS is slightly heavier than Hugo and Jekyll.

> Since of the benefits of styled-components is clean none duplicated CSS, it can be a good explaining why CSS size for these JavaScript SSGs are lower. One more hypothesis, old fashined SSGs use old fashion way for handling interactions and animations using CSS. JavaScript based use more JavaScript in ganaral, hence it might be used to replace functionlity that can be implemented with CSS.
> 
> {# TODO - "none duplicated"? not "non"? #}

Reviewer: ```suggestion
Since of the benefits of styled-components is clean, non-repetitive CSS, this could explain why CSS size for these JavaScript SSGs are lower. One more hypothesis is that old fashioned SSGs use old fashion methods for handling interactions and animations using CSS. JavaScript-based SSGs use more JavaScript in general, hence they might more often be used to replace functionality that could be implemented with CSS.
```
